### PR TITLE
Fix notebook export for already-open notebook documents

### LINKED ISSUE #7233: Export notebook doesn't put up the export quickpick for opening an existing notebook
1. Open a notebook
2. Reload VS Code. Notebook is still open
3. Click on the notebook toolbar export button
4. 🐛 export does nothing, quickpick isn't shown
5. Close and reopen notebook, then export works

Root cause is the notebookEditorProvider wasn't aware of notebookDocuments that were already open before the extension activated, so running the command does nothing because activeEditor is undefined and the editor list is empty

https://github.com/microsoft/vscode-jupyter/blob/b019cc9ce852a2de38df45f7db1beba3e503e6cb/src/client/datascience/commands/exportCommands.ts#L66-L70

https://github.com/microsoft/vscode-jupyter/blob/b019cc9ce852a2de38df45f7db1beba3e503e6cb/src/client/datascience/commands/exportCommands.ts#L89-L94

### PR DESCRIPTION
**Pull request by joyceerhl**
Fix #7233

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -731,6 +732,7 @@
                     "light": "resources/light/export_to_python.svg",
                     "dark": "resources/dark/export_to_python.svg"
                 },
+                "enablement": "notebookType == interactive",
```
Comment (joyceerhl) [inline package.json:735]: These commands were showing up in the command palette when they shouldn't

File: src/client/common/application/commands.ts [hunk 2]
```diff
@@ -143,7 +143,7 @@ export interface ICommandNameArgumentTypeMapping extends ICommandNameWithoutArgu
     [DSCommands.ExportToHTML]: [string | undefined, Uri | undefined, string | undefined, PythonEnvironment | undefined];
     [DSCommands.ExportToPDF]: [string | undefined, Uri | undefined, string | undefined, PythonEnvironment | undefined];
     [DSCommands.Export]: [string | undefined, Uri | undefined, string | undefined, PythonEnvironment | undefined];
-    [DSCommands.NativeNotebookExport]: [Uri | undefined];
+    [DSCommands.NativeNotebookExport]: [Uri | undefined | { notebookEditor: { notebookUri: Uri } }];
```
Comment (joyceerhl) [inline src/client/common/application/commands.ts:146]: The context that's passed from the notebookToolbar is not a simple Uri. Not sure if this was a recent change in VS Code. There might be other command handlers that are impacted by this.

File: src/client/datascience/notebook/notebookEditorProvider.ts [hunk 3]
```diff
@@ -76,6 +76,10 @@ export class NotebookEditorProvider implements INotebookEditorProvider {
         @inject(IExtensions) private readonly extensions: IExtensions
     ) {
         disposables.push(this);
+        // Register notebook documents that are already open e.g. after a workspace reload
+        for (const notebookDocument of this.vscodeNotebook.notebookDocuments) {
+            void this.onDidOpenNotebookDocument(notebookDocument);
+        }
```
Comment (joyceerhl) [inline src/client/datascience/notebook/notebookEditorProvider.ts:82]: This is the actual fix--notebookEditorProvider wasn't aware of notebookDocuments that were already loaded.

File: src/client/datascience/commands/exportCommands.ts [hunk 4]
```diff
@@ -63,9 +63,10 @@ export class ExportCommands implements IDisposable {
         this.disposables.push(disposable);
     }
 
-    private async nativeNotebookExport(uri?: Uri) {
-        const editor = uri
-            ? this.notebookProvider.editors.find((item) => this.fs.arePathsSame(item.file, uri))
+    private async nativeNotebookExport(uri?: Uri | { notebookEditor: { notebookUri: Uri } }) {
+        const notebookUri = uri !== undefined && 'notebookEditor' in uri ? uri.notebookEditor.notebookUri : uri;
```
Comment (DonJayamanne) [inline src/client/datascience/commands/exportCommands.ts:67]:
```suggestion
        const notebookUri = isUri(uri) ? uri : uri.notebookEditor.notebookUri;
```

Comment (DonJayamanne) [reply src/client/datascience/commands/exportCommands.ts:67]: You can use the helper `isUri` instead, you'd need to import that into this file.

**Review by DonJayamanne — APPROVED** [review]

**Comment by DonJayamanne** [comment]
@joyceerhl please could you update your code with main, this PR looks funky

File: src/client/datascience/notebook/notebookEditorProvider.ts (hunk 3 above)
Comment (DonJayamanne) [reply src/client/datascience/notebook/notebookEditorProvider.ts:82]: So we've always had this issue! Whoa

Comment (joyceerhl) [reply src/client/datascience/notebook/notebookEditorProvider.ts:82]: Actually, I debugged the release branch in VS Code 1.59.1, and it looks like VS Code has stopped firing `onDidOpenNotebookDocument` for already-open NotebookDocuments in Insiders, so previously this code wasn't even necessary. i.e. this was a behavior change in VS Code, not the Jupyter extension.

**Pull request merged**